### PR TITLE
Rudimentary test for manylinux_aarch64 with cibuildwheel

### PR DESCRIPTION
**Pull request by andyfaff**
Rudimentary test for manylinux_aarch64 during cibuildwheel

### CODE REVIEW

**Comment by andyfaff** [comment]
@rgommers, if you compare [5982358](https://github.com/scipy/scipy/pull/17011/commits/59823588fe01fde10f8e732342750b3966f2e751) and [babe664](https://github.com/scipy/scipy/pull/17011/commits/babe6647e6059c2b6d50a38ff309a6dd6ad65f4c) you'll see that `--oneline`` seems to be needed in order to get the wheel build to run. The latter works (with `--oneline`), but the former doesn't (without `--oneline`).

**Comment by andyfaff** [comment]
RIght, fixed the one-line thing correctly. I just search for the [wheel build] string in bash, which works.

**Comment by rgommers** [comment]
Thanks @andyfaff. Side note to start with: I forgot whether we wanted a separate `cibuildwheel` label for this or not. Otherwise it could be labelled `CI`.

**Review by rgommers — COMMENTED** [review]
This looks pretty good. The extra test time of 20 minutes seems okay to me. We could shorten it if it becomes a problem. The main thing I see is that a lot of test dependencies get installed. `cibuildwheel` takes those from `pyproject.toml`: https://cibuildwheel.readthedocs.io/en/stable/options/#dependency-versions. I think it makes sense to provide a separate file for that - all we need here is `pytest` and `threadpoolctl` I believe.

**Comment by andyfaff** [comment]
Squashed commits and force pushed to tidy things up. The last `[wheel build]` worked for aarch64 with the tests (trust me), so it's mergeable as is. I'll probably self-merge once the other tests complete (although they're unrelated).

**Comment by andyfaff** [comment]
The number of test dependencies is reduced for linux_aarch64, only pytest and threadpoolctl are installed. The only tests that are run are import of linalg/optimize, and the linalg test suite.

File: pyproject.toml [hunk 1]
```diff
@@ -142,14 +142,15 @@ dodoFile = "dev.py"
 [tool.cibuildwheel]
 skip = "cp36-* cp37-* pp* *_ppc64le *_i686 *_s390x *-musllinux*"
 build-verbosity = "3"
-before-test = "pip install -r {project}/tools/wheels/test_requirements.txt"
+# gmpy2 and scikit-umfpack are usually added for testing. However, there are
+# currently wheels missing that make the test script fail.
+test-requires = ["pytest", "pytest-cov", "pytest-xdist", "asv", "mpmath", "threadpoolctl", "pooch"]
```
Comment (rgommers) [inline pyproject.toml:147]: For when we touch this file again: we can also remove `pytest-cov` and `asv` here, they aren't used.

**Comment by rgommers** [comment]
Thanks @andyfaff, looks great!

File: .github/workflows/wheels.yml [hunk 2]
```diff
@@ -46,14 +46,18 @@ jobs:
         run: |
           set -xe
           COMMIT_MSG=$(git log --no-merges -1)
-          echo "::set-output name=message::$COMMIT_MSG"
+          RUN="0"
```
Comment (tylerjereddy) [inline .github/workflows/wheels.yml:49]: currently backporting this, though the maintenance branch may very well have the single line commit message check at this time

**Pull request merged**